test/cppsuite/test_harness/workload/thread_context.h

```diff
@@ -35,167 +35,131 @@
      */
-    bool
-    can_commit_transaction() const
+    void
+    set_commit_timestamp(WT_SESSION *session, const std::string &ts)
     {
-        return (_timestamp_manager->enabled() && _in_txn &&
-          (!_running || (_current_op_count > _max_op_count)));
+        std::string config = std::string(COMMIT_TS) + "=" + ts;
+        testutil_check(session->timestamp_transaction(session, config.c_str()));
     }
 
-    void
-    commit_transaction(WT_SESSION *session, const std::string &config)
+    /*
+     * op_count is the current number of operations that have been executed in the current
+     * transaction.
+     */
+    int64_t op_count = 0;
+
+    private:
+    /*
+     * _min_op_count and _max_op_count are the minimum and maximum number of operations within one
+     * transaction. is the current maximum number of operations that can be executed in the current
+     * transaction.
+     */
+    int64_t _min_op_count = 0;
+    int64_t _max_op_count = INT64_MAX;
+    int64_t _target_op_count = 0;
+    bool _in_txn = false;
+};
+
+/* Container class for a thread and any data types it may need to interact with the database. */
+class thread_context {
+    public:
+    thread_context(configuration *config, timestamp_manager *timestamp_manager,
+      workload_tracking *tracking, database &db)
+        : database(db), timestamp_manager(timestamp_manager), tracking(tracking),
+          transaction(transaction_context(config))
     {
-        /* A transaction cannot be committed if not started. */
-        testutil_assert(_in_txn);
-        testutil_check(
-          session->commit_transaction(session, config.empty() ? nullptr : config.c_str()));
-        _in_txn = false;
+        session = connection_manager::instance().create_session();
+        _throttle = throttle(config);
+
+        /* These won't exist for read threads which is why we use optional here. */
+        key_size = config->get_optional_int(KEY_SIZE, 1);
+        value_size = config->get_optional_int(VALUE_SIZE, 1);
+
+        testutil_assert(key_size > 0 && value_size > 0);
     }
 
     void
-    increment_operation_count(uint64_t inc = 1)
+    finish()
     {
-        _current_op_count += inc;
+        _running = false;
     }
 
-    /*
-     * Set a commit timestamp if the timestamp manager is enabled.
-     */
     void
-    set_commit_timestamp(WT_SESSION *session, wt_timestamp_t ts)
+    sleep()
     {
-        if (!_timestamp_manager->enabled())
-            return;
+        _throttle.sleep();
+    }
 
-        std::string config = std::string(COMMIT_TS) + "=" + _timestamp_manager->decimal_to_hex(ts);
-        testutil_check(session->timestamp_transaction(session, config.c_str()));
+    bool
+    running() const
+    {
+        return _running;
     }
 
+    WT_SESSION *session;
+    transaction_context transaction;
+    test_harness::timestamp_manager *timestamp_manager;
+    test_harness::workload_tracking *tracking;
+    test_harness::database database;
```

Comment:
I got annoyed at calling getters from the update, and read functions in the database operation structure.